testsuite/tests/kuadrant/authorino/metrics/conftest.py

```diff
@@ -0,0 +1,55 @@
+"""Conftest for the Authorino metrics tests"""
+from importlib import resources
+from datetime import datetime, timedelta
+
+import pytest
+
+from testsuite.openshift.objects.metrics import Metrics
+
+
+@pytest.fixture(scope="session")
+def metrics(openshift):
+    """
+    Return an instance of OpenShift metrics client
+    Skip tests if query route is not properly configured
+    """
+    # find thanos-querier route in the openshift-monitoring project
+    # this route allows to query metrics
+    openshift_monitoring = openshift.change_project("openshift-monitoring")
+    routes = openshift_monitoring.routes.for_service("thanos-querier")
+    if len(routes) > 0:
+        endpoint = ("https://" if "tls" in routes[0]["spec"] else "http://") + routes[0]["spec"]["host"]
+        return Metrics(endpoint, openshift.token, openshift.project)
+
+    return pytest.skip("Skipping metrics tests as query route is not properly configured")
+
+
+@pytest.fixture(scope="module")
+def label_metrics_service(authorino, module_label):
+    """Label Authorino controller-metrics service for the proper discovery"""
+    authorino.metrics_service.label({"app": module_label})
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module", autouse=True)
+def create_service_monitor(request, openshift, authorino, blame, module_label, metrics, label_metrics_service):
+    """Create ServiceMonitor object to follow Authorino /metrics and /server-metrics endpoints"""
+    with resources.path("testsuite.resources", "service-monitor.yaml") as path:
+        service_monitor_object = openshift.new_app(
+            path,
+            {
+                "NAME": blame("sm"),
+                "LABEL": module_label,
+            },
+        )
+    request.addfinalizer(service_monitor_object.delete)
+
+    # wait for the new configured endpoints to became available for metrics scraping
+    metrics.wait_for_scrape(authorino.metrics_service.name(), scrape_after=datetime.utcnow() + timedelta(seconds=90))
```

Comment:
you are using 90secs here, but the constant is for 30sec, why?

Reply:
There is some time before Prometheus starts to follow configured endpoints after creation of a new ServiceMonitor object. It is approximately 75 seconds. After, every scrape is performed every 30 seconds by default.